app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ShoppingListController < ApplicationController
+  def index
+    @user = current_user
+    @foods = Food.all
+    @recipes = Recipe.where(user_id: @user.id)
+    @recipe_foods = RecipeFood.where(recipe_id: @recipes.ids)
+    @shopping_list = []
+    @recipe_foods.each do |recipe_food|
+      @shopping_list << recipe_food
+    end
+    @final_list = @foods.where(id: @shopping_list.map(&:food_id))
+
+    @price = @final_list.pluck(:price)
+    @quantity = @final_list.pluck(:quantity)
```

Comment:
- [ ] [OPTIONAL] At the moment, the shopping list which you're generating is simply displaying the same quantity of food that you have available. The expected behaviour is that the shopping list should be calculated and generated based on the available quantity of food vs the actual quantity of food needed for the recipes. Please consider making this change in order to improve your project as this is only optional because the previous reviewer missed it👍